continuous-testing/use-cases/markmap.md

```diff
@@ -1,56 +1,134 @@
 * Continuous Testing
     * For Development APIs
-        * Unit Testing
-            * xUnit (e.g. Junit\[Java\], PyUnit\[Python\], NUnit\[C#\], CppUnit\[C++\]
-            * Tools
-                * Java
-                * Python
-                * C#
-                * C/C++
-        * Integration, Dependency and Stub Mocking
-            * Simulate dependencies
-            * Stub out coupled services for planned results
-            * Mock specific application behavior
-            * Tools
-                * Java
+        * For unit testing, integration, dependency and stub mocking
+            * For module testing
+                * Using xUnit
+                    * (Java) JUnit [&#x1F3C1; STARTER KIT](../starter-kits/#unit-teting-with-java)
+                    * (Python) PyUnit [&#x1F3C1; STARTER KIT](../starter-kits/#unit-teting-with-python)
+                    * (C#) NUnit
+                    * (C/C++) cUnit/CppUnit
+            * For mocking, stubbing and simulating test services
+                * Using Java
                     * Mockito
                     * EasyMock
                     * JMock
-                * Python
-                    * 
-                * C#
-                * C/C++
-        * Code Quality Analysis
-            * Semmle
+                * Using Python
+                    * PyTest
+                    * Nose2
+                    * Mock
+                * Using C#
+                    * Moq
+                    * NSubstitute
+                    * FakeItEasy
+                * Using C/C++
+                    * CMock
+                    * cpp-stub
+                    * googletest
+    * For Code Analysis
+        * For static analysis
+            * For formatting, standards and structure
+                * Using Java
+                    * Checkstyle
+                    * Spotbugs
+                    * PMD
+                * Using Python
+                    * PyLint
+                    * Pyflakes
+                    * PMD
+                * Using C#
+                    * StyleCop
+                    * Gendarme
+                    * PMD
+                * Using C/C++
+                    * CppCheck
+                    * Clang Static Analyzer / tidy
+                    * PMD
+        * For dynamic analysis
+            * For test coverage
+                * Using Java
+                    * Jacoco
+                    * Cobertura
+                * Using Python
+                    * Coverage
+                    * PyTest-cov
+                * Using C#
+                    * OpenCover
+                    * NCover
+                * Using C/C++
+                    * Gcov
+                    * lcov
+            * For complexity analysis and runtime performance
+                * Using Java
+                    * JProfiler
+                    * JMeter
+                    * JPF
+                * Using Python
+                    * Scalene
+                    * sys.settrace()/inspect/pdb
```

Comment:
Is this a library? If it's an internal Python library - then let's link to the documentation that describes this more fully.

Reply:
Linked.